src/helper.jl

```diff
@@ -32,24 +33,24 @@ using Base: significand_bits, exponent_bits, exponent_bias, exponent_mask, @pure
 # todo: check perf / type inference on 32 bit systems
 @inline function ldexpk(x::Float64, q::Integer)
     bias = exponent_bias(Float64)
-    expmax = exponent_max(Float64)
+    emax = exponent_max(Float64)
     shift = significand_bits(Float64)
     m = q >> 31
     m = (((m + q) >> 9) - m) << 7
     q = q - (m << 2)
     m += bias
     m = m < 0 ? 0 : m
-    m = m > expmax ? expmax : m
+    m = m > emax ? emax : m
     u = reinterpret(Float64, Int64(m) << shift)
     x = x * u * u * u * u
     u = reinterpret(Float64, Int64(q + bias) << shift)
     return x * u
 end
 
 @inline function ilogbp1(d::Float64)
-    m = d < 4.9090934652977266e-91
-    d = m ? 2.037035976334486e90 * d : d
-    q = ((reinterpret(Int64, d) >> 52) & 0x7ff) 
+    m = d < 4.9090934652977266e-91 #2.0^-300
+    d = m ? 2.037035976334486e90 * d : d  #2.0^300
```

Comment:
I spent some time thinking about this.
The `2f0^-64`, and `2f0^64` generally makes sense as a large/small value that will not under/overflow.
As they are both more or less the square root of of the smallest/largest value that does not underflow/overflow.

But this is less true for the Float64.
For 2.0^-300 and 2.0^300 are a fair bit under the square  root of the underflow/overflow flow amount (2.0^1024, and 2^-1024, having sqrt 512).}
The cube roots of the values is 2^341, 

```
julia> bin(reinterpret(Int64, 2.0^-300))
"10110100110000000000000000000000000000000000000000000000000000"
```
Doesn't tell me much about it being special.
It is not the cut off for denormal numbers or anything like that.